src/guide/customization.rs

```diff
@@ -0,0 +1,398 @@
+//! A tutorial on how to customize eyre by overriding the Context parameter
+//!
+//! # Introduction
+//!
+//! In this tutorial we will go through the process of implementing a custom context step by step:
+//!
+//! * Start with defining an empty context that captures nothing
+//! * Add support for capturing backtraces
+//!     * skip backtrace capture if our source captured one
+//! * Add support for a http status codes
+//!     * Implement a setter for Result<T, Report> with a new Trait
```

Comment:
It'd be nice to have anchor links here for the TOC, to jump to the relevant sections.